Classes/common/touchdb/TD_Database+Attachments.h

```diff
@@ -55,16 +58,16 @@ typedef enum { kTDAttachmentEncodingNone, kTDAttachmentEncodingGZIP } TDAttachme
                             encoding:(TDAttachmentEncoding *)outEncoding
                               status:(TDStatus *)outStatus;
 
-/** Returns the location of an attachment's file in the blob store. */
-- (NSString *)getAttachmentPathForSequence:(SequenceNumber)sequence
-                                     named:(NSString *)filename
-                                      type:(NSString **)outType
-                                  encoding:(TDAttachmentEncoding *)outEncoding
-                                    status:(TDStatus *)outStatus;
+/** Returns the blob for an attachment in the blob store. */
+- (id<CDTBlobReader>)getAttachmentBlobForSequence:(SequenceNumber)sequence
+                                            named:(NSString *)filename
+                                             type:(NSString **)outType
+                                         encoding:(TDAttachmentEncoding *)outEncoding
+                                           status:(TDStatus *)outStatus;
 
 /** Uses the "digest" field of the attachment dict to look up the attachment in the store and return
- * a file URL to it. DO NOT MODIFY THIS FILE! */
-- (NSURL *)fileForAttachmentDict:(NSDictionary *)attachmentDict;
+ * its blob. */
+- (id<CDTBlobReader>)blobForAttachmentDict:(NSDictionary *)attachmentDict;
```

Comment:
Why would someone do that? I mean, when you received an `id<SomeProtocol>`, you know that you should only use the methods defined in `SomeProtocol` because that is the documented behaviour. In other case, after the next update of the library, strange things may start to happen.

Also, I do not think we would solve this problem by diving the class. The method `TDBlobStore:pathForKey:` is not available in the public interface, however it is still possible to call it using `performSelector:withObject:`. Once you have the path to the attachment, you can create a `CDTBlobWriter` instance.